neo/Persistence/StoreDataCache.cs

```diff
@@ -0,0 +1,54 @@
+using Neo.IO;
+using Neo.IO.Caching;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Neo.Persistence
+{
+    internal class StoreDataCache<TKey, TValue> : DataCache<TKey, TValue>
+        where TKey : IEquatable<TKey>, ISerializable, new()
+        where TValue : class, ICloneable<TValue>, ISerializable, new()
+    {
+        private readonly IReadOnlyStore store;
+        private readonly ISnapshot snapshot;
+        private readonly byte prefix;
+
+        public StoreDataCache(IReadOnlyStore store, byte prefix)
+        {
+            this.store = store;
+            this.snapshot = store as ISnapshot;
```

Comment:
If is not?
```suggestion
            this.snapshot = store is ISnapshot ? (ISnapshot)store : null;
```

Reply:
The keyword `as` is for this case. Not need to change. If it is not `ISnapshot`, it returns `null`.